pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -569,17 +555,20 @@ func (c *Controller) registerHandlers(
 				})
 			},
 			UpdateFunc: func(old, cur interface{}) {
-				if !filter(old, cur) {
-					incrementEvent(otype, "update")
-					if !shouldEnqueue(otype, c.beginSync) {
+				if filter != nil {
+					if filter(old, cur) {
+						incrementEvent(otype, "updatesame")
 						return
 					}
-					c.queue.Push(func() error {
-						return wrappedHandler(cur, model.EventUpdate)
-					})
-				} else {
-					incrementEvent(otype, "updatesame")
```

Comment:
I still see this happening for endpoints on my cluster

Reply:
ah, because we have custom endpointsEqual filter. I see we kept the `updatesame`, I missed that earlier